File: api/src/main/java/com/czq/lbxx/api/entity/Role.java
package com.czq.lbxx.api.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Data;

@Entity
@Data
public class Role {

    @Id
    private Long id;

    private String cnName;

    private String jpName;

    private String avatar;

    private String cnActor;

    private String jpActor;

    @Column(columnDefinition = "TEXT")
    private String description;
}

File: web/api/types/role.ts
export interface Role {
  id: number;
  cnName: string;
  jpName: string;
  avatar: string;
  cnActor: string;
  jpActor: string;
  description: string;
  relationships: Relationship[];
  x?: number;
  y?: number;
  z?: number;
}

interface Relationship {
  targetId: number;
  // type: string;
}
